accounts-db/src/accounts_db.rs

```diff
@@ -3352,11 +3389,22 @@ impl AccountsDb {
         // Calculate store counts as if everything was purged
         // Then purge if we can
         let mut store_counts: HashMap<Slot, (usize, HashSet<Pubkey>)> = HashMap::new();
-        for (pubkey, (slot_list, ref_count)) in purges_zero_lamports.iter_mut() {
-            if purged_account_slots.contains_key(pubkey) {
-                *ref_count = self.accounts_index.ref_count_from_storage(pubkey);
-            }
-            slot_list.retain(|(slot, account_info)| {
+        for candidates_bin in candidates
+            .iter()
+            .filter(|bin| !bin.read().unwrap().is_empty())
```

Comment:
probably not necessary to check empty here. inner code can handle an empty

Reply:
i agree it looks like an optimization. But, there is zero contention on the lock. And, we don't expect empty lists anyway, so the check is unnecessary to avoid contention and adds extra work in the expected case. The code below with `iter_mut` will work just fine with a zero len map. It also just adds complexity for what seems to me to be no value.